using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Day09
{
    class Program
    {
        static void Main(string[] args)
        {
            //int res1 = ProblemOne("{}");
            //int res2 = ProblemOne("{{{}}}");
            //int res3 = ProblemOne("{{},{}}");
            //int res4 = ProblemOne("{{{},{},{{}}}}");
            //int res5 = ProblemOne("{<a>,<a>,<a>,<a>}");
            //int res6 = ProblemOne("{{<ab>},{<ab>},{<ab>},{<ab>}}");
            //int res7 = ProblemOne("{{<!!>},{<!!>},{<!!>},{<!!>}}");
            //int res8 = ProblemOne("{{<a!>},{<a!>},{<a!>},{<ab>}}");
            //
            //
            //
            //var res  = ProblemOne(Input);
            //ProblemTwo();

            int res1 = ProblemTwo("<>");
            int res2 = ProblemTwo("<random characters>");
            int res3 = ProblemTwo("<<<<>");
            int res4 = ProblemTwo("<{!>}>");
            int res5 = ProblemTwo("<!!>");
            int res6 = ProblemTwo("<!!!>>");
            int res7 = ProblemTwo("<{oAi!a,<{i<a>");

            int fin = ProblemTwo(Input);
        }

        static int ProblemOne(string input)
        {
            Stack<int> stack = new Stack<int>();
            bool garbage = false;
            int group = 0;
            int score = 0;
            int garbagecount = 0;
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == '!')
                {
                    //Skip next char
                    i++;
                    continue;
                }

                if (input[i] == '<')
                {
                    garbage = true;
                }

                if (input[i] == '>')
                {
                    garbage = false;
                }

                if (garbage)
                {
                    garbagecount++;
                }


                if (!garbage)
                {
                    if (input[i] == '{')
                    {
                        group++;
                    }

                    if (input[i] == '}')
                    {
                        score += group;
                        group--;
                    }
                }
            }

            return score;
        }

        static int ProblemTwo(string input)
        {
            Stack<int> stack = new Stack<int>();
            bool garbage = false;
            int group = 0;
            int score = 0;
            int garbagecount = 0;
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == '!')
                {
                    //Skip next char
                    i++;
                    continue;
                }

                if (input[i] == '<' && !garbage)
                {
                    garbage = true;
                    continue;
                }

                if (input[i] == '>' && garbage)
                {
                    garbage = false;
                    continue;
                }

                if (garbage)
                {
                    garbagecount++;
                }


                if (!garbage)
                {
                    if (input[i] == '{')
                    {
                        group++;
                    }

                    if (input[i] == '}')
                    {
                        score += group;
                        group--;
                    }
                }
            }

            return garbagecount;
        }

        private static string Example = @"{{<a!>},{<a!>},{<a!>},{<ab>}}";
        private static string Input = @"{{{{},{{{<a!!!>aa!!!>,,,}!>!>!!eo!!a>},<'!!!>,<!>}!!!>u!'!>,<!!!>!>o!!!>,<'!!!>!!!!<{{i>},{{}},{{<i!!!>u{!!eu!!>}}},{{{{{<a!>},<!o!>},<<!!!!!!,!!i>},{<{!>},<ii'>},{{<,o!!<'!!a!!!>>,{}}}}},{}},{{}}}},{{{{<e!e,!a'!!!>}ea}>,{<o!>,<{<!>},<<!<'!!ia>}},{{},<'!!!>!!e<a{u!>},<<!!!!!}>}},{{{{<!!!>!!!!iu!!!>!!!>{<eu}i,!!!!!!>}},{{<>,<!!!!o!aa{!>},<i!>,<!!u!!a!>},<a!i!>},<}au!>!!o>}},{<!>},<e}iaae'}!>!,!>o{,e!!{!>>}},{<aa!>!>a>,{<!{'u!!}{u,!>},<!>,<'}a>}},{{},{<}!!!>'o>}}},{{{<oi'<}!!!!>},<'!>},<u!>>},{<!!!!!!!>},<e!!!eu!>!o{'o,!!>,<{,oui'a!!ai!!!>},<<u}!!aaa!>,<>},{{{{<,'{'!>},<o}!>!>},<a!u!,!a!>},<a{>},<!>,<!{!>!!<oo}>}}}}},{{},{<<>},{}},{{{<!!!>},<,!'!!'u'!>},<!!ii!!!>i>},<!i!!}<'>},{{{<!!!!a!!!!a>}},{<!!!>aie!>>}},{{<}!!!>a!!i!>!!!!!>o!>,<>,{{<!<>}}}}},{{{{<{a!>,<!!!>!!u{!ao!>,<>,{<e!!o>}}}},{{{{<{e!!!!!!o{''!>,<>},{{{{},{<u!>},<!>,<!>>}},{}},{<>}},{{{<!!!>{!!o{}!!!>,,!>'ua'!!}!><!!!>>},{<!!!>ou!>},<!>ui,!!}{!!!>!>,<a{!!!>},<>}},{{<oa!!!>u!{u!''ae<e>},{{<uu!!!><!!!>>}}}}}},{{{{<!!!!{<}''a>,<}!>},<!>},<!>},<!!!>!!!!!!}!o!>},<}a>},{{{<e'aiau!>},<>,{<!!!>ie!!!>!!!>,<!!i<!!!>o!eu<'{>}},{<,a!!!>!>},<!!!!!>,a!>!!e!!!>},<ui!}>,{<!>},<aao'!>},<ea!!!>,<>}}},{{},{}}},{}},{{{<!'}!ei>}},{<!!!>,<!!u'}>,{<!>},<o}iu<!>>}}},{{{{}},<a{a!>},<!>!>},<i!>},<{}u>},{{{{{<e'!!u<!!!>oa<<!!u!>,<{a!oe!>,<!u>}},<ae!!!!!>!>},<!>,<!!!>{>}}}},{{{},{<a{oa}a<!!!>}>,{<!>},<!!!!!>ee{!!!!!>iu>}},{{<,!>a!!u!!u<a{!>},<!>},<,!a, i!!e{>},{<ue}e!!,oa!>},<a!>},<!!o>}}},{{{{<!>{!>,<!e'i!!<!!!!!e>},{{{}}}},{{<a!!'!!!>oo}}''!!!!!>!!!>},<ii''>},{{<!!!>u!!a!>!!'o!!!>,<i!>!!!!<',o}>}}}},{{{{},{{<}}'iau!!!>,<}!!!!{!e}!!ua!!!>>},{<,i!>},<!>{i!i{!o<aoe!!u!>i>}}}},{{{<<a!><>,<{e!,!><!!iaei!!e,<o!>},<!o>}},{{<}!o}!!i!>!!o}o!!!>oeia'!a>,{}},{<!>,<!>,<{o}ia!>},<!>!>},<aa!i'>},{<i!!e!!!>!!!>o!e>}},{{{{<{!!!>,<!>u''>},{}}},{{{{<}!{!>,<!>,<!!!>a!>},<}!>!!!>>},{{{<aao!>!a!!!!<!!i<i'a!!!uu>},{<i!>,<oiia,!>!o>}}}},{{{}},{<e!>!!!>uau!>,<}!!u!!!>!!!>!>,<eo>,<}!!!''a'a!!!>'!!!>!>},<e,,!>},<'<!>,<<>},{{{<u!>},<!u{!>,<!>},<!!a!>>},<e!{!!!!>}}},{{<a!!i!,'<',u,!!a'aa}!>>,<'a!>},<!>},<{aa,!<!!!!u!>!>,<!>,<!>!>,<<>},{{<>},{<a'!>!>},<'!uu,o!>,<<a!>,<!>},<!,!!!!>}},{{<i!ie!>},<!>!>,<!!ea,oi,!>'<<!!>,{{<>},{}}}}},{}},{{},{<!ao>,{<u!!,!!!>'i!!{ea!>a!>,<!!!>!a{>}},{<!>,<!!<'!!!!e!!'!>,<!!,!>,<<!>,<!!!>ua{>}},{{{},{{}}},{{<!>oia!>!>},<!!!>},<ia!!}>},{<a!!!>,<>}}},{{{{{<!!!>!!a!!!!,,a!!!>i}e{!>},<!!!!au!>,<o!>,<!!,>,<!!}!>,<!!a<!!}i!!!>u!>},<i!>,<!>>},{{{{<e<<{!>!a!>},<a}',}<e<!,>},<a!><}ua>}},<ea!!!>!>},<'o'<!uaua>}},<!e!!!!!!a}>}},{<!>},<!>{a>},{}}},{{{},<u<!!u!>>}},{{{{{<!,a'!}o!!!e!>!!eaee>}},<o!u<{!!u<!{!>!>a,{aa{}>},{{<}}!!!!!>!!!>,<a!>},<uuou,{,>},{<!>},<!!!!o!>},<!a!!!><e>}},{<!{!!u!ee!>,<!>,<!!,{{ia}aa!!,!>>,<>}},{{{{<!>,<!!!>e>}},{<a!!'iiaa<!a!!'o!>!>!!!>,<{>},{{{},{{<!!!!!!oa}<!>!e'ia!>!>},<{e}!!a!a!>,<o<>},{<'!}ae>}}}}},{{<!>},<}!>},<i!>!a!!!>,!!!>,<,!>},<e}>},{},{<>}},{{{{<e!>>,<!!!!!>},<!>,<ee{e'}>},{<!!e!!!><!>,<,!<{!>{>}}},{{<!!a!>},<!!!>,<,!!auu>}}},{{<!'oa!!!>!>,<e!>,<!!o{!!!!!!!>!!!!!!!>{!>},<>,<uoa!>},<!oe!'!!a>},{{<,!>!!>,<!!e<,}i,!!>},{{{<'e,<<}!!{!!io!>},<>}}},{{}}},{}}},{{{{{}}},{<}!!e!!}!>!>},<a!,'{!!!>>,{<!o>}}},{{{{{{{<,,{,ei!!!>},<'!>,<a!>,<u}!!u>},{<}!a!!!>,<ooe!!e!!ai}!!ai>}}}},<!>!>},<!>},<!!'>},{<!>},<!>!>,<<i!!!>!>a>},{{<!>,<{i!>},<!}!>'!!i{a!<>,<'!!!!!>,<a{eo'a{<>},{{},{}}}},{{{},{{{<oe!>,<!!!>e'!!!!!>}}{!ia!u>}},{{<!>,<,,!><!'}aa!!!>e!!!>>,<<!>,<!!!!'}!>},<a!!!>oa>}},{{{<!!!!{i!!a!!!>!!}!!!!!>e>},<!!!!!>{i!!!>e!'o!!e!!!>a'!!{!!u>},{{<!a!>a>},<}!!!>u!!!!!>aie>},{<,!o'!>,<a!>,<'>,{<!!!>!>>}}}},{<}<!!!uu>,{<ao}<!!!>,<}o!!!e!>au>}}},{},{{{<!!!>!>!!!!!>'!!!!!>!,a,o!uo,!>,<!>},<!>},<o>},{}},{<!!!>},<!>>,<au!!!>},<!u,!!!!!<{!>},<'!>,<!o<<i!!!!o>},{<>,{<ei!aa!'{!e!>},<{>}}},{{{<!!!e,u!>}!>},<!!!>,<!!!>,<!!!!a!>!!!!!>},<!!!>>}},{{<i}'{!!o,u{a!>},<a{iauu{>},{}},{{{{<o!!!>!!!a!>},<!a,!!!>}oae>,{<!>},<!!,>}},{}},{}}}}},{{{<'!>},<!!!>!>!!'}>}},{{<!!u!>,<'!!!!!!!>,<oe{a!>',!!!>u!>i<!o,>},{},{<}},!,'>,{{{<,a!{{!!u!!!>!!!>>}},<o>}}},{{<a!>},<e!>},<ua!!u<eui>}}},{{{{},{{{},<!>!!ie,ie}ao!>,<u>},{<,!!,aoa!>},<!!!><!!!>!!!>!>!o!>},<i!!!>i>}}},{<!>},<>,<e!e!!},}!!!>aa>},{<,i}!!!!,!>a!u'i!!!!!>u!>e}!>},<<!!!>},<>}}}},{{{<,!>,<ea{<e'!!{!!e!!!!!!!>!!o!>},<uai,!!,>,<!!i!>},<e!!!!u!a!!!>!>u!o!!!>},<<<>},{{{<,!,>}},{<!!!!!!!!a{!>,<u,e>}}},{<!>},<!>!>,<!!!>!a!!!!!<>,{<,'<e!!!!}o}{!!!>,<!!i<}e!!e>}},{<!!!>!!,!!ia{<a'>,{<}{!!'<{!>,<!>!>,<i!>},<!><!!!>>}}},{{<!a!>'!!!>!u!!!>,>}}},{}}},{{{}},{{{},<a!!!>{a!!,!!e'!>>},{}},{{<{<iao!!!!ai!!u!>,<a!>},<e>},{}}}},{{{<}!!,u',oa!!!><a!!,!>},<!!>},{<!>o!!o!>},<!>,<}!!!>>}},{<!!!!u}u!!!>>},{{},<!!!!!>},<'u!<!<!!<!>},<!!!<ae!>>}}},{{{<aoea!,!!!>a!>,<u!>a'{ao!!!>>,<i,!<iu!a<{,>},{<!>},<<i!>!!!>,e,}a!>},<!}!>!!ou!>,<u!!!!i>,{<!>,<!>},<oo},oo,!>a!!'<e>}}},{<o>}},{{{<i!!!aoo!>,<!e!>},<e,e!>},<!!!!!!!>o!>},<>},{<o!!i!!!!e!!{,!!!!!>},<,oi,!!u!!!>!!a>}},{<o{!>!!!>},<'}u!!!!!><!!!>!!!>>,<!>!!{!e<e!!!>!!!>uu>}}}}},{{{<o!>},<eaa!!!!!>i!!}e'}!!!!!>!!!>>},<o{>},{{}},{}},{{<!>,<!>},<a}>,<!>!!oa!>,<!!!!<,e>},{{{<!>},<a'iu'}u>},{{}}},{<oaa,,!>,<!!!!!>!>},<e!!a!!a}oe!!!>,>},{<!!!>},<!>},<',!}'}!>'!>,<!>},<a}u>}}}},{{},{},{<!>,<!>,<!>,<}i{!!!>,<<!!!>i>}},{{{{<{{!}iu!!{!!!!!!!>!!!>},<!!}u>},{{},{<!!a'u!>,<,!>!>},<,!!a!>},<!>!>},<ae'{>,<o!!{>}}},{{{{<!!!>ei!>,<o{'!!!>}!>},<!!!!!>},<!'>},{{<!!!>!{e,<>}}},{<ei,!!u<!!au>,{<aui!>!{!!!>},<ooa}u!>},<!!!>},<!>},<>}},{<<!>},<!>,<a<!>,<!>,<!>},<a>,{<!!!u!ai}!e!>o>}}}},{{<!>,<!!!!!u,'!>'!uoo>}}},{{<'e!i!>!>{u,!>!!{!>,<!!!>>}},{<}'}'a!>}!!!e!!i!!{<!>,<>,{<!>!>!>,<o!!!!ueo!!i>}}}},{{{<'>,{<!!!>u!<!>},<!u!!a!>i!!,!>oau,!!aa>}},{<ae<!>!!o!!ia!!!>!!!>>},{}},{{{<e!>{i{!!!!!!!!!!}!>,<!>,<o<{}>}},<!'e{i!>},<eea}<'!!!>!>'!,!ua}!!}>}},{}}},{{{{<<a!!!!{a!!i,a!!!>,,i!!!!!>o!a!!!>>}},{{}}},{{{<!!!><!>,<{,>}},{<{!>},<!!!>!!'!>},<a'e!'e}}{u',>,{{}}},{{<!!!!!!i!>a!>},<!!{!!i!!!>}o>,{<<e!!!>o!>},<!!!!!>!!a}a}!!u,!!!>!>,<>}}}},{{{},{{{<!>},<a>},{<,!!!>'u!!!>u!>},<<!>},<>}},{{<!>},<aaa<!{ea!>,<a!!!><,o!!!e!!<!!!>!>},<>}}}}}}},{{{{{{},<>}},{{<a{!>!>,<u!o!!}!>!aa!>!>},<}<,i!!<>}}}},{{{{<{!>!!!>>,{}}},{}},{{{},{}},{{<!>,<!>!>a!au<a!!u!!!a}!!<>},{<uaoa!!a!!!>{'e}!{!>},<}!>},<!!!!ou>}}}},{{{{{{{{<!>},<{!!!>,<i'{}!>,<uu!!!><!!{ao!!<!!iu>}},{{{<!>,<!,!>,<!>!!!>eao>},{<u!!!>{i!!!!!>}!>},<'!!}e''<a>,{<a!,!>},<!!o!!i!!!>!>},<e!!!>,{'!o!{!>},<!a!!!>,<>,<>}}},<{<e'auuaea''!>,<!>},<!>,<>}},<i{o!>},<i>},{{},<!>},<!>!>},<!!ie!>},<!e!>,<}'!>},<!>},<<a!>>}},{{<!>,<!>!!iie!>},<!>,<}!!aia!!{>}},{{<!>!>,<o!!!>a!,!>!!,a!!!>!>,<}iea>},<,!!!>!!!>!!a{i'!{u!!u,}!!!!!!',}!!,>}}},{{<i<a!,,!>},<!>},<!!a!>a>,{<a<!!uae<>}},{{<,!>},<}!u!!ea!>}>},{{<!!aa!}i!>,<,!i!!!>},<!>,<!!a,!!a>}}}},{{{<a!>},<!!!>>,{<!>>}},{{}},{<!!<!!!>uo!!o!aoe!>,<>}},{{{<!>,<!>!!ia'e>}},{{{{<o!>,<!!!>},<!!>},<!>,<!!}a,!!!>},<<!!o{aoiu!!a!!!>{>},{{},{}},{{},<}!'!!!!!!}!o!>,<u!>>}},{{<ai!>,<!>oai!!a!!!>'!!u!>,<>,{{},<!>{!!!!o,!>,<e!>},<i!!aa!!!>>}}}}},{{{{<!>,<!!a!!!>!!!!!>a!<!i>},{{<!!i',<!!u>}}}},{{{{{{<}i'!>!>},<,>}},{<<<a},ii!!u>}},{<}!>,<}!,!!o!>,<,eo<eaa<,!!!!!>,<{>}},{<>,<!!!>a>},{<u!>!!!>i!!i'!>,<ea!i>}}},{{},{}}}},{{{<!>,<!!<e>},<<!!<!!a{!!!>ii{,'!!iue>},{{{{{<!>,{u!!!>,<!a!a!>'!>,<u>},{{<o!!!!{!!!>ei!!,!!'!!a>},{{{},{}},{{<{iaa!!!!>},{{},<!!!>!>ii,!!e<!!!>},<>}}}},{<!>{}ea!!a}!>!o!>},<>}},{<}!>},<',!!!>o<u!>},<<{'!!!>a!!!>!!{!!!!{>}},{{{<!>!!!>a!!>},{<!>{!{,!>,<!!!>'>,<!>,<!!!!!!'!!eoa!!!>!!oo'>}}}},{{{},<!a!!oe>}}},{{<}!>},<<'!>,<<!!!>o{!uo>}}}}},{{{{{<!!!>!!},!!!>aai!a!>a,eo!>!}o>},<!!!>,<!>},<!>},<!!!!!!!>auu}!!!>},<!!!>},<!>},<''>},{{},{<{}!>,'!!!!!>},<!u{!!!>!!{!!!>a!!!>!!!>{!!!>},<>}},{{{{<!>,<a!>>,<,'e!!!>'oo,!!!i!>},<!<!>!!!>!!}a'>},<o<}{!>!!!!!>u<u!>,<i}!>,<>},{<}o{!>,<{<,!!{>,{}}},{{<!>>,{<!>},<!>},<ue!!!!<!e!!!><e}a!!!>}!>ie!>,<>}},{<<!>,<o!>},<ao<ea,>}}}},{{<<}!!!!!>!>},<a}!!ii<a'!eio!,>,<'!>!!!{'!!'a}i>}},{{{<',>},{}},{{}},{<!>},<<!a>}},{{<!!i!>},<'u!{{!>},<a!>,!!!>>,{{{<e!<a!!!>},<!a,,!!!!o!>u!!',>}}}},{{{<e!,!!!!!>'!!!>!!!!{,!!au!!!>!>aee<!>,<a!!!>>,<{!!!!!>!!<!,'oa!>,<!!o!!'!>,<!!>},{<!!oe}eu!>{!>a!>,<!>,<e>}},{<!!u!!e!!}ai!>,<!!!>},<<!!!>!>,<uu!i}'!>},<>,{<!!ee{{u>,{{{{<!>},<}i{!>!>,!!ao!!!>!!}>}}},{{{{},{}},<!!!>{!!!>!>},<<{aa!!!!!>,<>},<}<!!}>}}}}}}},{{{{{{<a'iu!!o<<a'u!!e{'!>},<>},<{!u!!!!!>!!uaaoa{!!!>},<{>},{{{<o!>},<}>,<!>,<{!>,<!!!>},<!!!>i!!ai'!>!>},<!>au!'>},{<!!!>a!>u!>},<!>!!{}{eu!!{e>}},{{},{<!>!>}u!a!!!>!e}!!!>,auaai!!!>!!>}}},{{{<eii!!,>}}}},{<!!!{>,{}},{<!>},<'eo<u'a'{!!!>!!!>u<'>}},{{{{},{{{<a!!}a>,{<!!!>!i!!e>}},{{},{<!>},<>}}},{<oua,!a!>,<!>!>,<!>,<,!!!i>,<!!e!,!!!!{e!!e,a!>},<!{{o'!>,<aa!>>},{{<<'i<!!!>}>},{}}},{{<!!{!!!>!i!>eea!ia!>}a!!''>},<!>},<'!>},<ao>}},{{{{{},{{<}!>,<u!!!!!!<!}!!!>>},{}}},{{<o!>},<o!>},<!e!!!>>,<e<'!!!!!>!!!!!!,a!>>}}},{<ie!!a}!>!>!a!>,<e'!>},<!>,aua!o,>}},{<!<}!>a!>},<<i!{u!>},<,!!!>}e{a!!!!<>},{{<}!>,<ea{!!!o!>,<!>,<!!!>eu!!<!!!e!>},<ia>}}},{{{<!>},<ai!!!>ii>}},{{{<{>},{<a!>},<!>,<'!>,<'!>,<!!o}!!!!!!a!>uaa!!e'>}},{<!!!a!!a!!!>>}},{<i!>},<!!!>eu<{u>}},{{<!!<!!!!<>,<uo!>,<>},{<!>},<e!!!!!!u!!!>>,<!!!{!!!>>},{<!!!>}i!>},<!'a{!>},<!'a}u!!{}!!!>,<>,{{<i!>!>,<u!>},<u!<e>}}}}},{{{{{<!>ua!>},<'u!!o!!!>!>,<>},<>},{},{{<au!!!>},<!!!>!!o!>>}}},{},{{<!!!>e}a!!!!!!!!!u!!'o'!>},<a>},<e}o!!!>!>},<!>{aue'ee<!>{a>}}},{{{<i{<o''!ea>,{{},{{<o>}}}}},{{{<}!'o!>},<iui>}}}},{{{{<''!!!!a}!!!>{!!o!!'!}!>i!!!!e<!{!!i>},<o>},{{{},<ie}!!!{!!!>!!>},{{{<!>},<!!}i>},{<,{!!!!!><e!!{!u<e!>!!!>},<!!a!>,<aa>}},<!aa>},{{<!!!>},<a!>,<!>},<!a!>},<!>,<>},{<a!>>}}},{<ea>}}}},{{<!!!>>},{<!e!!<<,!!!>aia!>},<e!!,>,{<!!!!!!!>u!!!o,<,!>a!!u!>a!>},<a!!!>,!>>,{{},<'!!!>'>}}},{{<>,{{<{,a!!i!!!>!!!>!>,!!u<!!!>!>>}}},{<ei<!><'ue!>i>,<!!ea!!{aa,>},{{<!>{!>,<!!o}ii!>},<!o'!>},<!>},<a,!!!{!>!>,<,>}}}}},{{{<!>},<i>},{<'<,a}!!!><o<!>},<!!ue}a!>},<!>{>}},{{{<ai!!{}!>},<,!!!>e!!!>},<!>!>!ei>,{<u!!e,<!!{o!>},<!!!!!>{a!!!},!!u!!>}},{<uou<a!!!>!!o!>u!>,<!!>},{{<!>},<i}{o!aa>},{<!<ie!!!>}'<!!{uauu!!!>!>>}}},{{{<ao'i{a<}ea{!!!>oa!!!!!!!>,{!>},<>},<!!>}},{}},{<!>},<i>,<!>,<ie!>o},!!u{!!i>}}},{{{{{},{}}},{{{{<i!>,<!}!>'}>}},{<!!!>!>,<o<!!!>ei!>},<oea!>a}>,{{},{<!,!!o!!!>,!!e!u!!!>!!{>}}}},{{{},{<oe!!e!>},<!!!a'i!!}{!>a!u!!ea!,{oi>}}}}},{{{<!>,<!>},<!>,<>,{<aoe<!>},<a!>},<!a}!!a!>},<>}}},{{{{{{<}u,!>,<},!>!>!eu!!!>u<!>},<!!}o!!>},<a!>,<!>,<ea}!!}ae}>}}},{{<!>},<a!>},<<a!>ae!,{u!!!>!!!>au!!!>>},<a!>},<a!!o!>},<a!>}>}},{{<,,e!!!>!>!!!>!!!>!}>},{<!>,<a}!>!>!>},<{!>},<!>},<!!!>!>!!a{>}},{{{{{}}},{<!a!!!>e!!!>!'a!{!e}i>}}}},{{},{<i,!!ao!>,<u<oi!>},<'!>,<>,{}},{<'i!!,a!>},<!!!>,<u}!!!!e{!>},<!>,}>}}},{{<oo!>,<a}!>,<!!e'a>},{{{{<o!aa}{<!!!>,e'a>},{<i!>,<{!!o}e!>},<!><>}}}}},{{{<}!>},<,!!o!!o<!>},<!!!>!!a!<}>,<a!!'{i!>},<i!>},<'i!>,!!i!>!>},<!><a>},{<,i!<au!>},<!!!>},<o!!!>!!!>e}!>,<,i>}},{{<o!>},<a'!!!>!!}!><a!>},<!<oo}{!>,<>},{<!!!!{!!a}}}{a',{au>,{}},{}},{{{{<ai!!!>,<!>},<,!!,<!},{<!>},<}!!!>'!!!>>},<!!!!!><!>},<!!!!!e!!i>},{{<!o,!>,<o!!!!!!a>},{<!>!>},<o!>},<!>ae<>,{<e,o}!!'{{!>!e<io!!!>!!!i{'!>!>,<>}}}},{{<ea'!!!!o}'a>},{{}},{}},{{{{<e!!!!!!!>!>>}},{<!!!>>}},{<'!>},<u!>'}>,<}>},{{{<!!}iu<!>,<!>,<!>},<!>,<>},{<'o{!!>}},{<i<}!>oa!>,<!>!!!>a!!!>!!uoa>}}},{{{{{{<e!oa!!!>!!,o!!!>>,{}},{<'!>{!!!>!!!>!>},<a<!>u!!u!!!!o!>>}},{{<e!!ao!<a!>},<!e!{>}}},<!!',iia}!!!><u<}>},{{<!!!>},<}}!>,<,!!!>{!!!''!!i!!a!>,<'>}},{}}}}}},{{{{<{{!!'!>!>,<<!!!>!>,<!!!>!<!>{!>},<!!{!>,<ia>},{<>}},{{<}<i!!}!!e!!<o!!!!u!!!>},<<!,a<>}},{{}}},{{{<,!!!><}!>!!!!!>!!oi{>},<{ueuio!!!>>},{{{},{}},{{<a!>},<!,!>!>,<!>!}!>},<aee'!>,!'oe!!!>>},{<!!!{''}i}o!!>}}},{{},<!>,<!aao<!<!>},<!!a>}}},{{{{},<,>},{}},{{{}}},{{{{{}},{{<!>},<!>,<a{!!>}}}},{{}},{{}}},{{<!!o!>},<!!!>},<!!!>!>e!>,>,{<o!!a!>o!>!!!!!>!>,<a!>a}!{!>,<!a!!!'>}},{{}}}}},{{{{{}},{<!{'a<!!!>}>,<u!'!>!>,<a!!!}!>a!>!>eee<!>,<>}},{{{{{},<!>,<{a!!{!>!>},<,!ueao'>},{{{<!!!!!>},<!!aa!>a<!>!,oo!!!>!!!>>}}},{{<>},<!!!>},<!>,<>}},{{{<e!!!>,<'!a,!!ae{!>,<'{a!>},<!uoeee>},{{{<oa!>},<!>,<!}!!!>>},{<a!{,'a!>},<!!,}oa'>}},{<<!!!!!>e}>},{<{a!<e{!!!!!{!>{'<!>},<!!oiua!!!>},<>,{<!!!><!>,<oo!!!>!!!>i!!!>''!!e'!!!>,<>}}}}},{{{<<!!'!>},<!!!>ia!>},<a!!a'!>>},{<o,{>}},{{{<'>},{<!!!>},<>}},{<a!>,<aae'a'<!>o!!!>,<>},{{<!>},<i!!!!<o!!!>},<!a>},{<>}}}},{{{{<a!>!>!>,<!,,!!<a!!!!!>e,!>{}!>,<!>,<!}>},<<!>},<{u<!!a!>,<!aaa!!!!e!!!!<aeou>},{{<oa>},{<}>}}},{{<!!!>i{>},<!>auue{>}}},{{{<!'a!,!!a!!!>!>,<<o,<!>ae,!!!>!e!!u,>},{}}},{{{}},{{<!>},<!>},<!>,<>,<,,o!>},<!!'},{!!a!>,>},{{{<<>},{<a!>,<!!!!!>!>!au!!!>!>,<!>,<!!!>u{au!>},<a!>},<>}},{<!u!>'!!!>auu>}}}},{{{{},{<a>}},{{},{{{<}oe>},{<u{}!!<}a!>,<}!!'!}!a!>a<>}}}},{{}}},{{<i>,<{o>},{<u!!!!i!!!>!>!!!>ia!!,{!!!>,<'a!}>},{{<oa!!!>!>},<!>},<!!>},{{},<u<u<!!!!!!!!!>},<'!!}!!!>e!>},<{!>,<>}}},{{{<aa!!!>'!>},<!{!>,<aa!!!!!>'>},{{}}},{{<o!!!!a{!>,<,au!!!><!!a,>},{}}}}}},{{{{{{<,a<oa!!!>'!!!>},<}!io>},{<>}}},{{<!!!>!!!o}!!!>!}a!> o!>}!>,<'a!!!!!'}!>},<!a>,{{<!a!!!>a,u!>,<>},{<!!!>},<!>,eooo!>},<>}}},{{{<!'i!!u!>!!!!!!!>aae!!!>o!!!>!i>}}}},{{{{{<!!!>o>,{<o,!>,<{u!aa!!}a,!!{!>,a}!!!>{>}},<!!eu!>!>},<,ei,e!>},<!!}ea!>>},{<e<uie,!>!!>,{{<{!!oa!>!!,u!!!>},<!!!!!!'!>},<!>,<,!!}a!>>},{<!>},<ue!!!>!!!!!>!!}>,<e!!}o}o!>},<oi{<!>,<e!'a<!!u>}}},{<u!!<u{!>},<!!!>!!!>{!io!!i>,<,{a!!!>},<a!,!!!><e!>o!!!',i!!o{!!!>},<!>>}},{{{<!!!!!!a!>!!'!!!!!>},<e{!!!>a!!!>!>},<!!!>!'}<>},{}}}}},{{{<!!!!u!>,!!!>,<oe,{ua{!>!>,<a!!!>a!>>},{},{{{{<}e}!>,<aai!!ii,>,{<!!!!!!!>!>},<!!a>}}},<!!'!!!>o>}}}}},{{{},{{},<i!>,<!>},<i}ua'<!>},<{>}},{<>,{}},{}},{{<u{!!!>},<,>},{<!!a}a,'!!!>!>i!!!e!>,<aia!!}!!!!!!!!{!>!!u!e>,{<,o!!!!!!!!a!''!e!><!!!!!>,<i!!a,!<!!{>}}},{{{{<!!!>!!,!!!>{}!!!>>},<!,e!o!!}ao!>,<>},{{},{<ao!!a!!!!!e<'a{!oa''{!!!>'>,{<o!><!>i!>!>,<'!>},<<!!<!u!iao!{,>}}}},{{{}},{<!!ae<>},{{<}ae>},{}}},{}}},{{{{<!{,i!!e!>!!!!!>!!i!!{ao!>},<,a{o>,{<!a!!e!>},<{e'u!>,<}!>},<'a!!u>}}},{{},<!!<!>>},{{{<'!>,<aao!!a!>e>},{<!>},<ai!!>}},{<a!<!!i!>!!<'!>,<'!!,<u!>!>},<a!}!>,<a>,{}}}},{{{<a,o<a!!!>}'!!!!}>},{<!>!!!>>}},{{{}},{<e'!>},<a{u!>},<!>i'<!!i!ui,!>},<a}>}},{<,!!!<o>}},{{{{<ai>},{}}}}}},{{{{<!!!!>}},{{<!!!>!!<u!',!>},<>},{<!!{!{>}}}}}}";

    }
}
