using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Day09
{
    class Program
    {
        static void Main(string[] args)
        {
            //int res1 = ProblemOne("{}");
            //int res2 = ProblemOne("{{{}}}");
            //int res3 = ProblemOne("{{},{}}");
            //int res4 = ProblemOne("{{{},{},{{}}}}");
            //int res5 = ProblemOne("{<a>,<a>,<a>,<a>}");
            //int res6 = ProblemOne("{{<ab>},{<ab>},{<ab>},{<ab>}}");
            //int res7 = ProblemOne("{{<!!>},{<!!>},{<!!>},{<!!>}}");
            //int res8 = ProblemOne("{{<a!>},{<a!>},{<a!>},{<ab>}}");
            //
            //
            //
            //var res  = ProblemOne(Input);
            //ProblemTwo();

            int res1 = ProblemTwo("<>");
            int res2 = ProblemTwo("<random characters>");
            int res3 = ProblemTwo("<<<<>");
            int res4 = ProblemTwo("<{!>}>");
            int res5 = ProblemTwo("<!!>");
            int res6 = ProblemTwo("<!!!>>");
            int res7 = ProblemTwo("<{oAi!a,<{i<a>");

            int fin = ProblemTwo(Input);
            var test = Parts1and2(Input);
        }


        public static (int, int) Parts1and2(string input)
        {
            int sum = 0, garbageCount = 0;

            int groupLevel = 0;
            bool inGarbage = false, skipNext = false;

            foreach (char c in input)
            {
                // If we should skip the next character in the
                // sequence. This will be a character that was
                // preceeded by a "!".
                if (skipNext)
                {
                    skipNext = false;
                    continue;
                }

                // Any character following a "!" should
                // be skipped.
                if (c == '!')
                {
                    skipNext = true;
                    continue;
                }

                // A "<" character marks the start of a garbage group.
                // If this character -is not- within a garbage group
                // then set the garbage group flag.
                // If this character -is- within a garbage group then
                // fall through and let this character be added to
                // the garbage character total.
                if (c == '<')
                {
                    if (inGarbage == false)
                    {
                        inGarbage = true;
                        continue;
                    }
                }

                // A ">" character marks the end of a garbage section.
                if (c == '>')
                {
                    inGarbage = false;
                    continue;
                }

                // Keep a track of the total characters that
                // are within a garbage section.
                if (inGarbage)
                {
                    garbageCount++;
                }

                // A "{" character that is not within a garbage
                // section will cause our nested group level
                // to increase. Each nested level will increase
                // the score for the group by one.
                if (c == '{' && !inGarbage)
                {
                    groupLevel++;
                }

                // A "}" character that is not within a garbage
                // section marks the end of the section.
                // Add the group score to our running total
                // and decrease the group level accordingly.
                if (c == '}' && !inGarbage)
                {
                    sum += groupLevel--;

                }
            }

            return (sum, garbageCount);
        }
        /*
        static int ProblemOne(string input)
        {
            Stack<int> stack = new Stack<int>();
            bool garbage = false;
            int group = 0;
            int score = 0;
            int garbagecount = 0;
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == '!')
                {
                    //Skip next char
                    i++;
                    continue;
                }

                if (input[i] == '<')
                {
                    garbage = true;
                }

                if (input[i] == '>')
                {
                    garbage = false;
                }

                if (garbage)
                {
                    garbagecount++;
                }


                if (!garbage)
                {
                    if (input[i] == '{')
                    {
                        group++;
                    }

                    if (input[i] == '}')
                    {
                        score += group;
                        group--;
                    }
                }
            }

            return score;
        }*/

static int ProblemTwo(string input)
{
    bool garbage = false;
    int group = 0;
    int score = 0;
    int garbagecount = 0;
    for (int i = 0; i < input.Length; i++)
    {
        switch (input[i])
        {
            case '!':
                i++;
                continue;
            case '<':
                if (!garbage)
                {
                    garbage = true;
                    continue;
                }
                break;
            case '>':
                garbage = false;
                continue;
        }
        
        if (garbage)
        {
            garbagecount++;
        }
        else
        {
            if (input[i] == '{')
            {
                group++;
            }

            if (input[i] == '}')
            {
                score += group;
                group--;
            }
        }
    }

    return garbagecount;
}

        private static string Example = @"{{<a!>},{<a!>},{<a!>},{<ab>}}";
        private static string Input = "{{{{},{{{<a!!!>aa!!!>,,,}!>!>!!eo!!a>},<'!!!>,<!>}!!!>u!'!>,<!!!>!>o!!!>,<'!!!>!!!!<{{i>},{{}},{{<i!!!>u{!!eu!!>}}},{{{{{<\"!>},<!o!>},<<!!!!!!,!!i>},{<{!>},<ii'>},{{<,o!!<'!!\"!!!>>,{}}}}},{}},{{}}}},{{{{<e!e,!a'!!!>}e\"}>,{<o!>,<{<!>},<<!<'!!ia>}},{{},<'!!!>!!e<\"{u!>},<<!!!!!}>}},{{{{<!!!>!!!!iu!!!>!!!>{<eu}i,!!!!!!>}},{{<>,<!!!!o!a\"{!>},<i!>,<!!u!!a!>},<\"!i!>},<}au!>!!o>}},{<!>},<e}iaae'}!>!,!>o{,e!!{!>>}},{<\"\"!>!>a>,{<!{'u!!}{u,!>},<!>,<'}a>}},{{},{<}!!!>'o>}}},{{{<oi'<}!!!!>},<'!>},<u!>>},{<!!!!!!!>},<e!!!eu!>!o{'o,!!>,<{,oui'\"!!\"i!!!>},<<u}!!\"aa!>,<>},{{{{<,'{'!>},<o}!>!>},<\"!u!,!a!>},<\"{>},<!>,<!{!>!!<oo}>}}}}},{{},{<<>},{}},{{{<!!!>},<,!'!!'u'!>},<!!ii!!!>i>},<!i!!}<'>},{{{<!!!!a!!!!a>}},{<!!!>aie!>>}},{{<}!!!>a!!i!>!!!!!>o!>,<>,{{<!<>}}}}},{{{{<{a!>,<!!!>!!u{!\"o!>,<>,{<e!!o>}}}},{{{{<{e!!!!!!o{''!>,<>},{{{{},{<u!>},<!>,<!>>}},{}},{<>}},{{{<!!!>{!!o{}!!!>,,!>'u\"'!!}!><!!!>>},{<!!!>ou!>},<!>ui,!!}{!!!>!>,<\"{!!!>},<>}},{{<oa!!!>u!{u!''ae<e>},{{<uu!!!><!!!>>}}}}}},{{{{<!!!!{<}''\">,<}!>},<!>},<!>},<!!!>!!!!!!}!o!>},<}a>},{{{<e'\"iau!>},<>,{<!!!>ie!!!>!!!>,<!!i<!!!>o!eu<'{>}},{<,a!!!>!>},<!!!!!>,a!>!!e!!!>},<ui!}>,{<!>},<aao'!>},<ea!!!>,<>}}},{{},{}}},{}},{{{<!'}!ei>}},{<!!!>,<!!u'}>,{<!>},<o}iu<!>>}}},{{{{}},<\"{\"!>},<!>!>},<i!>},<{}u>},{{{{{<e'!!u<!!!>o\"<<!!u!>,<{\"!oe!>,<!u>}},<ae!!!!!>!>},<!>,<!!!>{>}}}},{{{},{<\"{o\"}\"<!!!>}>,{<!>},<!!!!!>ee{!!!!!>iu>}},{{<,!>a!!u!!u<\"{!>},<!>},<,!\",i!!e{>},{<ue}e!!,oa!>},<\"!>},<!!o>}}},{{{{<!>{!>,<!e'i!!<!!!!!e>},{{{}}}},{{<\"!!'!!!>oo}}''!!!!!>!!!>},<ii''>},{{<!!!>u!!a!>!!'o!!!>,<i!>!!!!<',o}>}}}},{{{{},{{<}}'i\"u!!!>,<}!!!!{!e}!!u\"!!!>>},{<,i!>},<!>{i!i{!o<\"oe!!u!>i>}}}},{{{<<a!><>,<{e!,!><!!iaei!!e,<o!>},<!o>}},{{<}!o}!!i!>!!o}o!!!>oeia'!\">,{}},{<!>,<!>,<{o}ia!>},<!>!>},<\"\"!i'>},{<i!!e!!!>!!!>o!e>}},{{{{<{!!!>,<!>u''>},{}}},{{{{<}!{!>,<!>,<!!!>a!>},<}!>!!!>>},{{{<a\"o!>!a!!!!<!!i<i'\"!!!uu>},{<i!>,<oii\",!>!o>}}}},{{{}},{<e!>!!!>uau!>,<}!!u!!!>!!!>!>,<eo>,<}!!!''a'a!!!>'!!!>!>},<e,,!>},<'<!>,<<>},{{{<u!>},<!u{!>,<!>},<!!\"!>>},<e!{!!!!>}}},{{<\"!!i!,'<',u,!!\"'aa}!>>,<'a!>},<!>},<{aa,!<!!!!u!>!>,<!>,<!>!>,<<>},{{<>},{<a'!>!>},<'!uu,o!>,<<\"!>,<!>},<!,!!!!>}},{{<i!ie!>},<!>!>,<!!e\",oi,!>'<<!!>,{{<>},{}}}}},{}},{{},{<!\"o>,{<u!!,!!!>'i!!{ea!>\"!>,<!!!>!a{>}},{<!>,<!!<'!!!!e!!'!>,<!!,!>,<<!>,<!!!>ua{>}},{{{},{{}}},{{<!>oia!>!>},<!!!>},<i\"!!}>},{<a!!!>,<>}}},{{{{{<!!!>!!\"!!!!,,a!!!>i}e{!>},<!!!!au!>,<o!>,<!!,>,<!!}!>,<!!\"<!!}i!!!>u!>},<i!>,<!>>},{{{{<e<<{!>!\"!>},<\"}',}<e<!,>},<a!><}ua>}},<ea!!!>!>},<'o'<!u\"u\">}},<!e!!!!!!\"}>}},{<!>},<!>{a>},{}}},{{{},<u<!!u!>>}},{{{{{<!,\"'!}o!!!e!>!!e\"ee>}},<o!u<{!!u<!{!>!>\",{a\"{}>},{{<}}!!!!!>!!!>,<\"!>},<uuou,{,>},{<!>},<!!!!o!>},<!a!!!><e>}},{<!{!!u!ee!>,<!>,<!!,{{ia}a\"!!,!>>,<>}},{{{{<!>,<!!!>e>}},{<a!!'iiaa<!a!!'o!>!>!!!>,<{>},{{{},{{<!!!!!!oa}<!>!e'ia!>!>},<{e}!!a!\"!>,<o<>},{<'!}\"e>}}}}},{{<!>},<}!>},<i!>!a!!!>,!!!>,<,!>},<e}>},{},{<>}},{{{{<e!>>,<!!!!!>},<!>,<ee{e'}>},{<!!e!!!><!>,<,!<{!>{>}}},{{<!!\"!>},<!!!>,<,!!\"uu>}}},{{<!'o\"!!!>!>,<e!>,<!!o{!!!!!!!>!!!!!!!>{!>},<>,<uo\"!>},<!oe!'!!a>},{{<,!>!!>,<!!e<,}i,!!>},{{{<'e,<<}!!{!!io!>},<>}}},{{}}},{}}},{{{{{}}},{<}!!e!!}!>!>},<\"!,'{!!!>>,{<!o>}}},{{{{{{{<,,{,ei!!!>},<'!>,<\"!>,<u}!!u>},{<}!\"!!!>,<ooe!!e!!ai}!!\"i>}}}},<!>!>},<!>},<!!'>},{<!>},<!>!>,<<i!!!>!>\">},{{<!>,<{i!>},<!}!>'!!i{\"!<>,<'!!!!!>,<a{eo'a{<>},{{},{}}}},{{{},{{{<oe!>,<!!!>e'!!!!!>}}{!ia!u>}},{{<!>,<,,!><!'}\"a!!!>e!!!>>,<<!>,<!!!!'}!>},<a!!!>o\">}},{{{<!!!!{i!!\"!!!>!!}!!!!!>e>},<!!!!!>{i!!!>e!'o!!e!!!>\"'!!{!!u>},{{<!\"!>\">},<}!!!>u!!!!!>\"ie>},{<,!o'!>,<\"!>,<'>,{<!!!>!>>}}}},{<}<!!!uu>,{<\"o}<!!!>,<}o!!!e!>au>}}},{},{{{<!!!>!>!!!!!>'!!!!!>!,a,o!uo,!>,<!>},<!>},<o>},{}},{<!!!>},<!>>,<au!!!>},<!u,!!!!!<{!>},<'!>,<!o<<i!!!!o>},{<>,{<ei!aa!'{!e!>},<{>}}},{{{<!!!e,u!>}!>},<!!!>,<!!!>,<!!!!a!>!!!!!>},<!!!>>}},{{<i}'{!!o,u{\"!>},<\"{iauu{>},{}},{{{{<o!!!>!!!\"!>},<!\",!!!>}o\"e>,{<!>},<!!,>}},{}},{}}}}},{{{<'!>},<!!!>!>!!'}>}},{{<!!u!>,<'!!!!!!!>,<oe{\"!>',!!!>u!>i<!o,>},{},{<}},!,'>,{{{<,a!{{!!u!!!>!!!>>}},<o>}}},{{<\"!>},<e!>},<u\"!!u<eui>}}},{{{{},{{{},<!>!!ie,ie}ao!>,<u>},{<,!!,aoa!>},<!!!><!!!>!!!>!>!o!>},<i!!!>i>}}},{<!>},<>,<e!e!!},}!!!>aa>},{<,i}!!!!,!>a!u'i!!!!!>u!>e}!>},<<!!!>},<>}}}},{{{<,!>,<e\"{<e'!!{!!e!!!!!!!>!!o!>},<u\"i,!!,>,<!!i!>},<e!!!!u!\"!!!>!>u!o!!!>},<<<>},{{{<,!,>}},{<!!!!!!!!a{!>,<u,e>}}},{<!>},<!>!>,<!!!>!a!!!!!<>,{<,'<e!!!!}o}{!!!>,<!!i<}e!!e>}},{<!!!>!!,!!i\"{<\"'>,{<}{!!'<{!>,<!>!>,<i!>},<!><!!!>>}}},{{<!a!>'!!!>!u!!!>,>}}},{}}},{{{}},{{{},<a!!!>{\"!!,!!e'!>>},{}},{{<{<i\"o!!!!\"i!!u!>,<\"!>},<e>},{}}}},{{{<}!!,u',oa!!!><\"!!,!>},<!!>},{<!>o!!o!>},<!>,<}!!!>>}},{<!!!!u}u!!!>>},{{},<!!!!!>},<'u!<!<!!<!>},<!!!<ae!>>}}},{{{<\"oea!,!!!>\"!>,<u!>a'{\"o!!!>>,<i,!<iu!a<{,>},{<!>},<<i!>!!!>,e,}\"!>},<!}!>!!ou!>,<u!!!!i>,{<!>,<!>},<oo},oo,!>\"!!'<e>}}},{<o>}},{{{<i!!!aoo!>,<!e!>},<e,e!>},<!!!!!!!>o!>},<>},{<o!!i!!!!e!!{,!!!!!>},<,oi,!!u!!!>!!a>}},{<o{!>!!!>},<'}u!!!!!><!!!>!!!>>,<!>!!{!e<e!!!>!!!>uu>}}}}},{{{<o!>},<eaa!!!!!>i!!}e'}!!!!!>!!!>>},<o{>},{{}},{}},{{<!>,<!>},<a}>,<!>!!o\"!>,<!!!!<,e>},{{{<!>},<a'iu'}u>},{{}}},{<oaa,,!>,<!!!!!>!>},<e!!a!!\"}oe!!!>,>},{<!!!>},<!>},<',!}'}!>'!>,<!>},<\"}u>}}}},{{},{},{<!>,<!>,<!>,<}i{!!!>,<<!!!>i>}},{{{{<{{!}iu!!{!!!!!!!>!!!>},<!!}u>},{{},{<!!a'u!>,<,!>!>},<,!!\"!>},<!>!>},<ae'{>,<o!!{>}}},{{{{<!!!>ei!>,<o{'!!!>}!>},<!!!!!>},<!'>},{{<!!!>!{e,<>}}},{<ei,!!u<!!au>,{<\"ui!>!{!!!>},<oo\"}u!>},<!!!>},<!>},<>}},{<<!>},<!>,<\"<!>,<!>,<!>},<a>,{<!!!u!ai}!e!>o>}}}},{{<!>,<!!!!!u,'!>'!uoo>}}},{{<'e!i!>!>{u,!>!!{!>,<!!!>>}},{<}'}'a!>}!!!e!!i!!{<!>,<>,{<!>!>!>,<o!!!!ueo!!i>}}}},{{{<'>,{<!!!>u!<!>},<!u!!a!>i!!,!>o\"u,!!\"a>}},{<\"e<!>!!o!!ia!!!>!!!>>},{}},{{{<e!>{i{!!!!!!!!!!}!>,<!>,<o<{}>}},<!'e{i!>},<eea}<'!!!>!>'!,!ua}!!}>}},{}}},{{{{<<\"!!!!{a!!i,a!!!>,,i!!!!!>o!a!!!>>}},{{}}},{{{<!!!><!>,<{,>}},{<{!>},<!!!>!!'!>},<a'e!'e}}{u',>,{{}}},{{<!!!!!!i!>a!>},<!!{!!i!!!>}o>,{<<e!!!>o!>},<!!!!!>!!a}\"}!!u,!!!>!>,<>}}}},{{{},{{{<!>},<a>},{<,!!!>'u!!!>u!>},<<!>},<>}},{{<!>},<\"\"\"<!{ea!>,<\"!!!><,o!!!e!!<!!!>!>},<>}}}}}}},{{{{{{},<>}},{{<\"{!>!>,<u!o!!}!>!a\"!>!>},<}<,i!!<>}}}},{{{{<{!>!!!>>,{}}},{}},{{{},{}},{{<!>,<!>!>\"!au<\"!!u!!!a}!!<>},{<uao\"!!\"!!!>{'e}!{!>},<}!>},<!!!!ou>}}}},{{{{{{{{<!>},<{!!!>,<i'{}!>,<uu!!!><!!{\"o!!<!!iu>}},{{{<!>,<!,!>,<!>!!!>e\"o>},{<u!!!>{i!!!!!>}!>},<'!!}e''<\">,{<\"!,!>},<!!o!!i!!!>!>},<e!!!>,{'!o!{!>},<!a!!!>,<>,<>}}},<{<e'auuae\"''!>,<!>},<!>,<>}},<i{o!>},<i>},{{},<!>},<!>!>},<!!ie!>},<!e!>,<}'!>},<!>},<<\"!>>}},{{<!>,<!>!!iie!>},<!>,<}!!aia!!{>}},{{<!>!>,<o!!!>a!,!>!!,a!!!>!>,<}ie\">},<,!!!>!!!>!!\"{i'!{u!!u,}!!!!!!',}!!,>}}},{{<i<a!,,!>},<!>},<!!a!>\">,{<\"<!!uae<>}},{{<,!>},<}!u!!ea!>}>},{{<!!\"a!}i!>,<,!i!!!>},<!>,<!!\",!!\">}}}},{{{<\"!>},<!!!>>,{<!>>}},{{}},{<!!<!!!>uo!!o!aoe!>,<>}},{{{<!>,<!>!!ia'e>}},{{{{<o!>,<!!!>},<!!>},<!>,<!!}\",!!!>},<<!!o{aoiu!!a!!!>{>},{{},{}},{{},<}!'!!!!!!}!o!>,<u!>>}},{{<\"i!>,<!>o\"i!!\"!!!>'!!u!>,<>,{{},<!>{!!!!o,!>,<e!>},<i!!a\"!!!>>}}}}},{{{{<!>,<!!a!!!>!!!!!>\"!<!i>},{{<!!i',<!!u>}}}},{{{{{{<}i'!>!>},<,>}},{<<<a},ii!!u>}},{<}!>,<}!,!!o!>,<,eo<ea\"<,!!!!!>,<{>}},{<>,<!!!>a>},{<u!>!!!>i!!i'!>,<e\"!i>}}},{{},{}}}},{{{<!>,<!!<e>},<<!!<!!a{!!!>ii{,'!!iue>},{{{{{<!>,{u!!!>,<!a!\"!>'!>,<u>},{{<o!!!!{!!!>ei!!,!!'!!a>},{{{},{}},{{<{i\"a!!!!>},{{},<!!!>!>ii,!!e<!!!>},<>}}}},{<!>{}e\"!!a}!>!o!>},<>}},{<}!>},<',!!!>o<u!>},<<{'!!!>a!!!>!!{!!!!{>}},{{{<!>!!!>\"!!>},{<!>{!{,!>,<!!!>'>,<!>,<!!!!!!'!!eo\"!!!>!!oo'>}}}},{{{},<!a!!oe>}}},{{<}!>},<<'!>,<<!!!>o{!uo>}}}}},{{{{{<!!!>!!},!!!>a\"i!\"!>\",eo!>!}o>},<!!!>,<!>},<!>},<!!!!!!!>auu}!!!>},<!!!>},<!>},<''>},{{},{<{}!>,'!!!!!>},<!u{!!!>!!{!!!>\"!!!>!!!>{!!!>},<>}},{{{{<!>,<\"!>>,<,'e!!!>'oo,!!!i!>},<!<!>!!!>!!}\"'>},<o<}{!>!!!!!>u<u!>,<i}!>,<>},{<}o{!>,<{<,!!{>,{}}},{{<!>>,{<!>},<!>},<ue!!!!<!e!!!><e}\"!!!>}!>ie!>,<>}},{<<!>,<o!>},<\"o<ea,>}}}},{{<<}!!!!!>!>},<\"}!!ii<\"'!eio!,>,<'!>!!!{'!!'\"}i>}},{{{<',>},{}},{{}},{<!>},<<!a>}},{{<!!i!>},<'u!{{!>},<a!>,!!!>>,{{{<e!<\"!!!>},<!\",,!!!!o!>u!!',>}}}},{{{<e!,!!!!!>'!!!>!!!!{,!!\"u!!!>!>\"ee<!>,<\"!!!>>,<{!!!!!>!!<!,'o\"!>,<!!o!!'!>,<!!>},{<!!oe}eu!>{!>a!>,<!>,<e>}},{<!!u!!e!!}\"i!>,<!!!>},<<!!!>!>,<uu!i}'!>},<>,{<!!ee{{u>,{{{{<!>},<}i{!>!>,!!ao!!!>!!}>}}},{{{{},{}},<!!!>{!!!>!>},<<{a\"!!!!!>,<>},<}<!!}>}}}}}}},{{{{{{<a'iu!!o<<\"'u!!e{'!>},<>},<{!u!!!!!>!!u\"ao\"{!!!>},<{>},{{{<o!>},<}>,<!>,<{!>,<!!!>},<!!!>i!!ai'!>!>},<!>\"u!'>},{<!!!>\"!>u!>},<!>!!{}{eu!!{e>}},{{},{<!>!>}u!\"!!!>!e}!!!>,au\"\"i!!!>!!>}}},{{{<eii!!,>}}}},{<!!!{>,{}},{<!>},<'eo<u'\"'{!!!>!!!>u<'>}},{{{{},{{{<\"!!}\">,{<!!!>!i!!e>}},{{},{<!>},<>}}},{<ou\",!a!>,<!>!>,<!>,<,!!!i>,<!!e!,!!!!{e!!e,a!>},<!{{o'!>,<\"\"!>>},{{<<'i<!!!>}>},{}}},{{<!!{!!!>!i!>eea!ia!>}a!!''>},<!>},<'!>},<ao>}},{{{{{},{{<}!>,<u!!!!!!<!}!!!>>},{}}},{{<o!>},<o!>},<!e!!!>>,<e<'!!!!!>!!!!!!,\"!>>}}},{<ie!!\"}!>!>!\"!>,<e'!>},<!>,\"u\"!o,>}},{<!<}!>\"!>},<<i!{u!>},<,!!!>}e{a!!!!<>},{{<}!>,<ea{!!!o!>,<!>,<!!!>eu!!<!!!e!>},<i\">}}},{{{<!>},<ai!!!>ii>}},{{{<{>},{<\"!>},<!>,<'!>,<'!>,<!!o}!!!!!!a!>u\"\"!!e'>}},{<!!!a!!\"!!!>>}},{<i!>},<!!!>eu<{u>}},{{<!!<!!!!<>,<uo!>,<>},{<!>},<e!!!!!!u!!!>>,<!!!{!!!>>},{<!!!>}i!>},<!'a{!>},<!'a}u!!{}!!!>,<>,{{<i!>!>,<u!>},<u!<e>}}}}},{{{{{<!>ua!>},<'u!!o!!!>!>,<>},<>},{},{{<au!!!>},<!!!>!!o!>>}}},{},{{<!!!>e}\"!!!!!!!!!u!!'o'!>},<a>},<e}o!!!>!>},<!>{aue'ee<!>{a>}}},{{{<i{<o''!e\">,{{},{{<o>}}}}},{{{<}!'o!>},<iui>}}}},{{{{<''!!!!\"}!!!>{!!o!!'!}!>i!!!!e<!{!!i>},<o>},{{{},<ie}!!!{!!!>!!>},{{{<!>},<!!}i>},{<,{!!!!!><e!!{!u<e!>!!!>},<!!a!>,<\"\">}},<!aa>},{{<!!!>},<a!>,<!>},<!a!>},<!>,<>},{<a!>>}}},{<e\">}}}},{{<!!!>>},{<!e!!<<,!!!>\"ia!>},<e!!,>,{<!!!!!!!>u!!!o,<,!>a!!u!>a!>},<\"!!!>,!>>,{{},<'!!!>'>}}},{{<>,{{<{,a!!i!!!>!!!>!>,!!u<!!!>!>>}}},{<ei<!><'ue!>i>,<!!ea!!{\"\",>},{{<!>{!>,<!!o}ii!>},<!o'!>},<!>},<\",!!!{!>!>,<,>}}}}},{{{<!>},<i>},{<'<,a}!!!><o<!>},<!!ue}\"!>},<!>{>}},{{{<ai!!{}!>},<,!!!>e!!!>},<!>!>!ei>,{<u!!e,<!!{o!>},<!!!!!>{\"!!!},!!u!!>}},{<uou<a!!!>!!o!>u!>,<!!>},{{<!>},<i}{o!\"a>},{<!<ie!!!>}'<!!{u\"uu!!!>!>>}}},{{{<ao'i{a<}e\"{!!!>oa!!!!!!!>,{!>},<>},<!!>}},{}},{<!>},<i>,<!>,<ie!>o},!!u{!!i>}}},{{{{{},{}}},{{{{<i!>,<!}!>'}>}},{<!!!>!>,<o<!!!>ei!>},<oe\"!>\"}>,{{},{<!,!!o!!!>,!!e!u!!!>!!{>}}}},{{{},{<oe!!e!>},<!!!a'i!!}{!>\"!u!!ea!,{oi>}}}}},{{{<!>,<!>},<!>,<>,{<aoe<!>},<\"!>},<!\"}!!a!>},<>}}},{{{{{{<}u,!>,<},!>!>!eu!!!>u<!>},<!!}o!!>},<\"!>,<!>,<ea}!!}ae}>}}},{{<!>},<a!>},<<\"!>\"e!,{u!!!>!!!>au!!!>>},<a!>},<\"!!o!>},<a!>}>}},{{<,,e!!!>!>!!!>!!!>!}>},{<!>,<\"}!>!>!>},<{!>},<!>},<!!!>!>!!\"{>}},{{{{{}}},{<!a!!!>e!!!>!'a!{!e}i>}}}},{{},{<i,!!ao!>,<u<oi!>},<'!>,<>,{}},{<'i!!,\"!>},<!!!>,<u}!!!!e{!>},<!>,}>}}},{{<oo!>,<a}!>,<!!e'a>},{{{{<o!\"a}{<!!!>,e'a>},{<i!>,<{!!o}e!>},<!><>}}}}},{{{<}!>},<,!!o!!o<!>},<!!!>!!a!<}>,<a!!'{i!>},<i!>},<'i!>,!!i!>!>},<!><\">},{<,i!<au!>},<!!!>},<o!!!>!!!>e}!>,<,i>}},{{<o!>},<\"'!!!>!!}!><\"!>},<!<oo}{!>,<>},{<!!!!{!!\"}}}{a',{au>,{}},{}},{{{{<ai!!!>,<!>},<,!!,<!},{<!>},<}!!!>'!!!>>},<!!!!!><!>},<!!!!!e!!i>},{{<!o,!>,<o!!!!!!a>},{<!>!>},<o!>},<!>ae<>,{<e,o}!!'{{!>!e<io!!!>!!!i{'!>!>,<>}}}},{{<ea'!!!!o}'a>},{{}},{}},{{{{<e!!!!!!!>!>>}},{<!!!>>}},{<'!>},<u!>'}>,<}>},{{{<!!}iu<!>,<!>,<!>},<!>,<>},{<'o{!!>}},{<i<}!>o\"!>,<!>!!!>\"!!!>!!uoa>}}},{{{{{{<e!o\"!!!>!!,o!!!>>,{}},{<'!>{!!!>!!!>!>},<\"<!>u!!u!!!!o!>>}},{{<e!!ao!<a!>},<!e!{>}}},<!!',iia}!!!><u<}>},{{<!!!>},<}}!>,<,!!!>{!!!''!!i!!\"!>,<'>}},{}}}}}},{{{{<{{!!'!>!>,<<!!!>!>,<!!!>!<!>{!>},<!!{!>,<i\">},{<>}},{{<}<i!!}!!e!!<o!!!!u!!!>},<<!,a<>}},{{}}},{{{<,!!!><}!>!!!!!>!!oi{>},<{ueuio!!!>>},{{{},{}},{{<a!>},<!,!>!>,<!>!}!>},<aee'!>,!'oe!!!>>},{<!!!{''}i}o!!>}}},{{},<!>,<!\"\"o<!<!>},<!!\">}}},{{{{},<,>},{}},{{{}}},{{{{{}},{{<!>},<!>,<\"{!!>}}}},{{}},{{}}},{{<!!o!>},<!!!>},<!!!>!>e!>,>,{<o!!a!>o!>!!!!!>!>,<a!>\"}!{!>,<!\"!!!'>}},{{}}}}},{{{{{}},{<!{'\"<!!!>}>,<u!'!>!>,<a!!!}!>a!>!>eee<!>,<>}},{{{{{},<!>,<{a!!{!>!>},<,!ueao'>},{{{<!!!!!>},<!!aa!>a<!>!,oo!!!>!!!>>}}},{{<>},<!!!>},<!>,<>}},{{{<e!!!>,<'!a,!!ae{!>,<'{a!>},<!uoeee>},{{{<oa!>},<!>,<!}!!!>>},{<\"!{,'a!>},<!!,}oa'>}},{<<!!!!!>e}>},{<{\"!<e{!!!!!{!>{'<!>},<!!oiua!!!>},<>,{<!!!><!>,<oo!!!>!!!>i!!!>''!!e'!!!>,<>}}}}},{{{<<!!'!>},<!!!>i\"!>},<\"!!\"'!>>},{<o,{>}},{{{<'>},{<!!!>},<>}},{<a!>,<\"\"e'a'<!>o!!!>,<>},{{<!>},<i!!!!<o!!!>},<!\">},{<>}}}},{{{{<\"!>!>!>,<!,,!!<\"!!!!!>e,!>{}!>,<!>,<!}>},<<!>},<{u<!!\"!>,<!a\"\"!!!!e!!!!<\"eou>},{{<o\">},{<}>}}},{{<!!!>i{>},<!>\"uue{>}}},{{{<!'a!,!!\"!!!>!>,<<o,<!>ae,!!!>!e!!u,>},{}}},{{{}},{{<!>},<!>},<!>,<>,<,,o!>},<!!'},{!!a!>,>},{{{<<>},{<\"!>,<!!!!!>!>!\"u!!!>!>,<!>,<!!!>u{au!>},<\"!>},<>}},{<!u!>'!!!>\"uu>}}}},{{{{},{<a>}},{{},{{{<}oe>},{<u{}!!<}a!>,<}!!'!}!\"!>\"<>}}}},{{}}},{{<i>,<{o>},{<u!!!!i!!!>!>!!!>i\"!!,{!!!>,<'\"!}>},{{<oa!!!>!>},<!>},<!!>},{{},<u<u<!!!!!!!!!>},<'!!}!!!>e!>},<{!>,<>}}},{{{<\"a!!!>'!>},<!{!>,<aa!!!!!>'>},{{}}},{{<o!!!!\"{!>,<,\"u!!!><!!a,>},{}}}}}},{{{{{{<,a<o\"!!!>'!!!>},<}!io>},{<>}}},{{<!!!>!!!o}!!!>!}\"!>o!>}!>,<'a!!!!!'}!>},<!a>,{{<!a!!!>\",u!>,<>},{<!!!>},<!>,eooo!>},<>}}},{{{<!'i!!u!>!!!!!!!>aae!!!>o!!!>!i>}}}},{{{{{<!!!>o>,{<o,!>,<{u!\"\"!!}\",!!{!>,\"}!!!>{>}},<!!eu!>!>},<,ei,e!>},<!!}e\"!>>},{<e<uie,!>!!>,{{<{!!oa!>!!,u!!!>},<!!!!!!'!>},<!>,<,!!}\"!>>},{<!>},<ue!!!>!!!!!>!!}>,<e!!}o}o!>},<oi{<!>,<e!'\"<!!u>}}},{<u!!<u{!>},<!!!>!!!>{!io!!i>,<,{\"!!!>},<a!,!!!><e!>o!!!',i!!o{!!!>},<!>>}},{{{<!!!!!!a!>!!'!!!!!>},<e{!!!>\"!!!>!>},<!!!>!'}<>},{}}}}},{{{<!!!!u!>,!!!>,<oe,{ua{!>!>,<\"!!!>a!>>},{},{{{{<}e}!>,<a\"i!!ii,>,{<!!!!!!!>!>},<!!\">}}},<!!'!!!>o>}}}}},{{{},{{},<i!>,<!>},<i}ua'<!>},<{>}},{<>,{}},{}},{{<u{!!!>},<,>},{<!!\"}a,'!!!>!>i!!!e!>,<aia!!}!!!!!!!!{!>!!u!e>,{<,o!!!!!!!!\"!''!e!><!!!!!>,<i!!a,!<!!{>}}},{{{{<!!!>!!,!!!>{}!!!>>},<!,e!o!!}\"o!>,<>},{{},{<ao!!a!!!!!e<'\"{!o\"''{!!!>'>,{<o!><!>i!>!>,<'!>},<<!!<!u!iao!{,>}}}},{{{}},{<!!ae<>},{{<}ae>},{}}},{}}},{{{{<!{,i!!e!>!!!!!>!!i!!{\"o!>},<,a{o>,{<!a!!e!>},<{e'u!>,<}!>},<'\"!!u>}}},{{},<!!<!>>},{{{<'!>,<a\"o!!a!>e>},{<!>},<\"i!!>}},{<\"!<!!i!>!!<'!>,<'!!,<u!>!>},<a!}!>,<a>,{}}}},{{{<a,o<\"!!!>}'!!!!}>},{<!>!!!>>}},{{{}},{<e'!>},<a{u!>},<!>i'<!!i!ui,!>},<\"}>}},{<,!!!<o>}},{{{{<ai>},{}}}}}},{{{{<!!!!>}},{{<!!!>!!<u!',!>},<>},{<!!{!{>}}}}}}";
    }
}
